ReactSkia/components/RSkComponentScrollView.cpp

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "include/core/SkPaint.h"
+#include "include/core/SkPictureRecorder.h"
+
+#include "compositor/layers/PictureLayer.h"
+
+#include "ReactSkia/components/RSkComponentScrollView.h"
+#include "ReactSkia/utils/RnsUtils.h"
+#include "ReactSkia/views/common/RSkDrawUtils.h"
+
+namespace facebook {
+namespace react {
+
+#define SCROLL_NEXT 1
+#define SCROLL_PREV 0
+
+using namespace RSkDrawUtils;
+
+RSkComponentScrollView::RSkComponentScrollView(const ShadowView &shadowView , RSkSurfaceWindow * surface)
+    : RSkComponent(shadowView),
+      containerWidth_(shadowView.layoutMetrics.frame.size.width),
+      containerHeight_(shadowView.layoutMetrics.frame.size.height),
+      surface_(surface) {}
+
+
+void RSkComponentScrollView::updateComponentProps(
+    const ShadowView &newShadowView,
+    bool forceUpadate) {
+
+  auto const &scrollViewProps = *std::static_pointer_cast<ScrollViewProps const>(newShadowView.props);
+
+  isScrollEnabled_ = scrollViewProps.scrollEnabled;
+
+  if( forceUpadate || (contentInset_ != scrollViewProps.contentInset)) {
+       contentInset_ = scrollViewProps.contentInset;
+       forceBitmapReset_ = true;
+  }
+
+}
+
+/* ScrollView childrens are maintained locally by component */
+/* ScrollView childrens are not added to render layer tree */
+void RSkComponentScrollView::mountChildComponent(
+    std::shared_ptr<RSkComponent> newChildComponent,
+    const int index) {
+
+    RNS_LOG_INFO("ScrollView mountChildComponent tag["<< newChildComponent->getComponentData().tag << "] index["<<index << "]");
+
+    childList_.insert(childList_.begin() + index, newChildComponent);
+    forceBitmapReset_ = true;
+
+    Rect childFrame = newChildComponent->getComponentData().layoutMetrics.frame;
+    if((childFrame.origin.y + childFrame.size.height) > containerHeight_)
```

Comment:
Is that the calculation intended to exclude/reduce the Edge inset set. If then won't it be  minus  contentInset value.
If I understood correctly, getContentFrame(), will this helps ?

Reply:
this logic is to get the container width & height .
We have two frames here 
1) ScrollView frame  : which is visible area for displaying scrolling content. This is received from the layout metrics frame. 
2) Scrolling Content frame : which is the container where the content is available. This value will vary based on how many childrens we have .

Reply:
Thanks for the detailed explanation. The need for adding  contentInset_ is where i'm not clear.